api/app/controllers/webhook.controller.ts

```diff
@@ -154,3 +165,171 @@ export async function create(req: Request, res: Response) {
+
+    if (!jsonContent) {
+      throw new Error(
+        "Malformed data: transcript content might not be in the correct format"
+      );
+    }
+
+    const transcriptHash = generateUniqueHash(jsonContent);
+    const totalWords = getTotalWords(jsonContent.body);
+    const content = jsonContent;
+
+    if (!transcriptHash || !totalWords) {
+      throw new Error(
+        "Malformed data: transcript content might not be in the correct format"
+      );
+    }
+
+    const existingTranscript = await Transcript.findOne({
+      where: { transcriptUrl: rawUrl },
+    });
+
+    if (existingTranscript && type === "added") {
+      throw new Error("transcript already exists");
+    }
+
+    if (!isTranscriptValid(transcript_by)) {
+      throw new Error(
+        "Transcript not from TSTBTC or does not need review - did not queue transcript"
+      );
+    }
+
+    const transcript: TranscriptAttributes = {
+      originalContent: {
+        ...content,
+        title: content.title.trim(),
+      },
+      content: content,
+      transcriptHash,
+      transcriptUrl: rawUrl,
+      status: TranscriptStatus.queued,
+      contentTotalWords: totalWords,
+    };
+
+    let transcriptData: any;
+
+    if (type === "added" || (type === "modified" && !existingTranscript)) {
+      transcriptData = await Transcript.create(transcript);
+    } else if (type === "modified" && existingTranscript) {
+      // Update the transcript in the database
+      await existingTranscript.update(transcript);
+
+      // Invalidate the cache
+      const redisTransaction = redis.multi();
+      redisTransaction.del(`transcript:${existingTranscript.id}`);
+      await redisTransaction.exec();
+
+      transcriptData = existingTranscript;
+    }
+
+    // Send alert to Discord
+    await sendAlert(
+      `Transcript Queued Successfully`,
+      transcriptData.originalContent.title,
+      transcriptData.transcriptUrl,
+      transcriptData.transcriptHash
+    );
+
+    const redisNewTranscriptTransaction = redis.multi();
+
+    redisNewTranscriptTransaction.sadd("cachedTranscripts", transcriptData.id);
+
+    redisNewTranscriptTransaction.set(
+      `transcript:${transcriptData.id}`,
+      JSON.stringify(transcriptData),
+      "EX",
+      CACHE_EXPIRATION
+    );
+
+    for (let i = 0; i < PAGE_COUNT; i++) {
+      redisNewTranscriptTransaction.del(`transcriptsPage:${i}`);
+    }
+
+    await redisNewTranscriptTransaction.exec((err, _results) => {
+      if (err) {
+        Transcript.destroy({ where: { id: transcriptData.id } });
+        throw err;
+      }
+    });
+  }
+}
+
+function isValidEnvironmentAndBranch(branch: string, env: string): boolean {
+  const allowedBranches = ["master", "staging", "development"];
+  if (!allowedBranches.includes(branch)) {
+    return false;
+  }
+
+  return (
+    (branch === "master" && env === "production") ||
+    (branch === "staging" && env === "staging") ||
+    (branch === "development" && env === "development")
```

Comment:
```suggestion
    (branch === "dev" && env === "development")
```

Reply:
We will have to notify others testing with this